### ISSUE TITLE
change isolation level of event fetch to ReadCommitted

### ISSUE THREAD

**Issue by tirumaraiselvan** [hasura/graphql-engine#6041]
Since the [query that is run](https://github.com/hasura/graphql-engine/blob/4c3890b3eadde0f7c04b03885c182c85a098516f/server/src-lib/Hasura/Eventing/EventTrigger.hs#L417) is of the form:

```
SELECT FOR UPDATE .. SKIP LOCKED
UPDATE ..
```

Even with concurrent queries running in ReadCommitted isolation level, this should be safe from re-deliveries. The first transaction to lock the event will "win" and others will ignore that event.

**Comment by tirumaraiselvan** [comment]
This should fix errors of the following type which are seen in fairly active event trigger usage:

```
{"type":"event-trigger","timestamp":"2020-08-31T17:08:47.158+0000","level":"error","detail":{"internal":{"statement":"\n UPDATE hdb_catalog.event_log\n SET locked = 't'\n WHERE id IN ( SELECT l.id\n FROM hdb_catalog.event_log l\n WHERE l.delivered = 'f' and l.error = 'f' and l.locked = 'f'\n and (l.next_retry_at is NULL or l.next_retry_at <= now())\n and l.archived = 'f'\n ORDER BY created_at\n LIMIT $1\n FOR UPDATE SKIP LOCKED )\n RETURNING id, schema_name, table_name, trigger_name, payload::json, tries, created_at\n ","prepared":true,"error":{"exec_status":"FatalError","hint":null,"message":"could not serialize access due to concurrent update","status_code":"40001","description":null},"arguments":["(Oid 20,Just (\"\\NUL\\NUL\\NUL\\NUL\\NUL\\NUL\\NULd\",Binary))"]},"path":"$","error":"database query error","code":"unexpected"}​} 
```

**Comment by tirumaraiselvan** [comment]
closed by #6120
